### PR TITLE
Feature+Refactor - A whole bunch of features and changes

### PR DESCRIPTION
**Pull request by PKWadsy**
This PR contains the following improvements:

- Reorganises tests
- Adds absolute imports
- Updates ZodValidator to work with any time which conforms to a certain shape. This can be made even more generic but I think we can leave it for now.
- Tests for specific error types to be thrown. Closes https://github.com/PilloFoundation/duvet/issues/50
- Adds a new testing script test:unit to run tests with coverage on non-integration tests.
- Coverage now includes all src files, not just files which were tested
- Makes middleware building type safe and more readable by adding a new `HandlerBuilder` interface and two implementing classes: `MiddlewareHandlerBuiler` and `BaseHandlerBuilder`
- Significant updates to the validator architecture. Each validator must now be a dictionary of keys to validator functions.
- The request and response type has been made generic throughout the codebase. For testing, many test just set these to the `object` type. Others test with express specifically. Closes issue #14 
- Created a fully fledged express wrapper response type which supports files, headers, cookies etc and a function which converts this object to operations on the express response type.
- Fixed up many tests
- Removed the example as it was causing too much trouble/confusion. Anything that was in there should either be in a separate examples repo or the tests folder.
- Moved express related things into a separate express folder.
- Moved and renamed some files.
- Other small fixes.

The reason I have made so many changes in a single PR is if I had made each change off develop, there would have been horrible merge conflicts to deal with in each of them (especially because of the renaming and use of generic request and response types).

As I was developing, one change would lead to breaking other aspects of the codebase. e.g. using the generic request and response types would break Zod validation (for some reason). Rather than come up with a hack, I thought it would be better to just fix it there and then. Same logic applies for why many other changes were grouped into this PR.

TODO:

- [ ] Consider what the default library should expose (perhaps there should be default wrappers on things)
- [ ] Write more extensive tests.
- [x] Move tests which are testing the Express specific parts of the framework into a folder for that
- [x] Rename zod validator to express zod validator OR make a generic zod validator

### CODE REVIEW

File: src/express/sendWrappedExpressResponse.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
+// sendResponse.ts
+import { Response } from "express";
+import { ExpressResponseWrapper } from "./models/ExpressResponseWrapper";
+
+/**
+ * Sends a wrapped express response.
+ * @param wrappedExpressResponse The wrapped express response to send.
+ * @param expressResponse The express response object to send the response with.
+ */
+export function sendWrappedExpressResponse(
```
Comment (joshuabrownenz) [inline src/express/sendWrappedExpressResponse.ts:10]: Amazing job with this. With this being such a critical component I think we need to delegate a fair amount of testing resources to this. If you don't want to do it now. Please make an issue to track.

**Review by joshuabrownenz — CHANGES_REQUESTED** [review]
Minimal changes. Please see comments.
Biggest one I want to draw attention to is whether you've made the zodValidator to narrow.

Can you also add "undef" and "middlewares" to allowed spelling words.

Chuurs

File: src/core/endpoint-tree/fs-builder/applyFileToEndpointTree.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+    // Check if the endpoint is a Duvet endpoint
+    if (isDuvetExport(duvetExport) !== true) {
+      throw new Error(`Endpoint at route ${filePath} is not a Duvet endpoint`);
+    }
+
+    const endpoint = duvetExport.data;
+
+    if (isDuvetEndpoint(endpoint) !== true) {
+      throw new Error(`Endpoint at route ${filePath} is not a Duvet endpoint`);
```
[the remark this answers is not in the document]
Comment (PKWadsy) [reply src/core/endpoint-tree/fs-builder/applyFileToEndpointTree.ts:60]: This is addressed in #48

File: src/express/sendWrappedExpressResponse.ts (hunk 1 above)
Comment (PKWadsy) [reply src/express/sendWrappedExpressResponse.ts:10]: Cheers mate. Now made an issue #51

File: src/zod/zodValidator.ts [hunk 3]
```diff
@@ -15,60 +15,42 @@ type RequestFields = "body" | "params" | "query" | "headers" | "cookies";
 function zodRequestFieldValidator<
```
[the remark this answers is not in the document]
Comment (PKWadsy) [reply src/zod/zodValidator.ts:15]: Funnily enough, this is also addressed in #48

**Pull request merged**